portable/NetworkInterface/xilinx_ultrascale/uncached_memory.c

```diff
@@ -69,11 +69,18 @@
 #include "uncached_memory.h"
 
 #if ( ipconfigULTRASCALE == 1 )
-    /* Reserve 2 MB of memory. */
-    #define uncMEMORY_SIZE         0x200000U
-    #define DDR_MEMORY_END         ( XPAR_PSU_DDR_0_S_AXI_HIGHADDR )
-    #define uncMEMORY_ATTRIBUTE    NORM_NONCACHE | INNER_SHAREABLE
-#else
+    #if defined( __aarch64__ ) || defined( ARMA53_32 )
+        /* Reserve 2 MB of memory. */
+        #define uncMEMORY_SIZE         0x200000U
```

Comment:
It would be nice if it also works when `uncMEMORY_SIZE` is predefined, so make each of the three \#define's conditional:
~~~diff
-    #define uncMEMORY_SIZE         0x200000U
+    #ifndef uncMEMORY_SIZE
+        #define uncMEMORY_SIZE         0x200000U
+    #endif
~~~
The module `uncached_memory.c` is also used by other software, for instance for data caching of an SD-card. In that case `uncMEMORY_SIZE` may become bigger.